web/src/compose.js

```diff
@@ -789,3 +795,24 @@ export function initialize() {
         }
     }
 }
+
+export function reset_compose_scheduling_state() {
+    $("#compose-textarea").prop("disabled", false);
+    $("#schedule-confirm").hide();
+    popover_menus.reset_selected_schedule_time();
+    $("#compose-send-button").show();
+    $("#compose-textarea").removeAttr("data-scheduled-message-id");
+}
+
+function schedule_message_to_custom_date() {
+    const request = create_message_object();
+    const selected_send_later_time = popover_menus.get_selected_send_later_time();
+    request.content = `/schedule ${selected_send_later_time}\n` + request.content;
```

Comment:
Why does this have `/schedule` at the start -- this isn't a slash command as far as I know?

Reply:
Oh, sigh, it is doing a slash command internally. OK, we'll want to rewrite that -- in this code path we should be able to just actually pass parameters to a function like `schedule_message` without slash command parsing/unparsing; can be a follow-up refactor though.

Reply:
Okay.